### ISSUE TITLE
Fix S4457: False positive when ArgumentException thrown after awaited call

### ISSUE THREAD

**Issue by sm-g** [SonarSource/sonar-csharp#1819]
```cs
        public async Task Handle()
        {
            var customers = await _loader.GetAllCustomers();

            if (customers.Count == 3)
                throw new ArgumentException("oops");
        }
```

7.4.0.6452
VS 15.8.1

**Comment by Evangelink** [comment]
Hi @sm-g,

I am a bit bothered by your code example. I agree that the rule shouldn't report here but at the same time I think it's really weird to throw a `ArgumentException` in such case because it has nothing to do with an argument being passed. If you replace this exception type by another you will see that the error is no longer reported.

I am assuming you have created a simpler case for us to see the issue, isn't it?

Thanks for the feedback anyway :)

**Comment by sm-g** [comment]
Yes, sample just to illustrate. Maybe that is poor design of exceptions - throw ArgumentException after argument was validated via awaited db call - but S4457 is incorrect.
